core/src/consensus.rs

```diff
@@ -1042,46 +1043,86 @@ impl Tower {
         self.last_switch_threshold_check.is_none()
     }
 
-    /// Performs threshold check for `slot`
-    ///
-    /// If it passes the check returns None, otherwise returns Some(fork_stake)
-    pub fn check_vote_stake_threshold(
+    /// Checks a single vote threshold for `slot`
+    fn check_vote_stake_threshold(
+        vote: Option<&Lockout>,
+        vote_state: &VoteState,
+        threshold_depth: usize,
+        threshold_size: f64,
+        slot: Slot,
+        voted_stakes: &HashMap<Slot, u64>,
+        total_stake: u64,
+    ) -> ThresholdDecision {
+        let vote = match vote {
+            None => return ThresholdDecision::PassedThreshold,
+            Some(vote) => vote,
+        };
+        let fork_stake = match voted_stakes.get(&vote.slot()) {
+            None => {
+                // We haven't seen any votes on this fork yet, so no stake
+                return ThresholdDecision::FailedThreshold(threshold_depth as u64, 0);
+            }
+            Some(fork_stake) => fork_stake,
+        };
+
+        let lockout = *fork_stake as f64 / total_stake as f64;
+        trace!(
+            "fork_stake slot: {}, vote slot: {}, lockout: {} fork_stake: {} total_stake: {}",
+            slot,
+            vote.slot(),
+            lockout,
+            fork_stake,
+            total_stake
+        );
+        if vote.confirmation_count() as usize > threshold_depth {
+            for old_vote in &vote_state.votes {
```

Comment:
I think this needs to be 
```suggestion
            for old_vote in &self.vote_state.votes {
```
as `vote_state` is the state after `process_slot_vote_unchecked` was called.

Reply:
`&self.vote_state` is what we passed into this function for `vote_state`. It should be the vote state from before `process_slot_vote_unchecked`